### ISSUE TITLE
Upgrade to 2.0.0-preview

### ISSUE THREAD

**Issue by Kagamine** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#282]

**Comment by ensleep** [comment]
What is the progress of this issue?I think I really need it to support dotnet core 2.0.

**Comment by caleblloyd** [comment]
I want to check it out but the instructions [here](https://www.microsoft.com/net/core/preview#linuxubuntu) say

```
Before you start, please remove any previous versions of .NET Core from your system
```

I have a lot of projects that rely on 1.x. Is 2.0 backwards compatible with 1? If so I can give this a shot.

**Comment by caleblloyd** [comment]
Entity Framework Core 2.0 will only target .NET Standard 2.0

https://github.com/aspnet/Announcements/issues/246

> This means you will be able to use Entity Framework Core 2.0 on:
> 
> - .NET Framework 4.6.1 and up
> - .NET Core 2.x
> - Mono 5.x

This means that in-turn, `Pomelo.EntityFrameworkCore.MySql` will only target .NET Standard 2.0 and have the same framework requirements

**Comment by labiraus** [comment]
The Sapient Guardian team say that there are going to be significant issues upgrading from 1.1 to 2.0.
How doable do you think this is going to be?
https://github.com/SapientGuardian/SapientGuardian.EntityFrameworkCore.MySql/issues/56

**Comment by caleblloyd** [comment]
@labiraus it's doable.  I'll have a better timeline in a day or two, I'm going to start taking an initial look at this now.

We'll have a 2.0.0 RTM a couple weeks after upstream entity framework does.  I'd like to get a preview out on nuget within a couple weeks.

**Comment by caleblloyd** [comment]
So it is going to be a fair amount of work to update to `2.0.0-preview`.  Off the bat I see a few namespaces with a lot of changes:

- `Extensions`
- `Storage/Internal`
- `Scaffolding`

I don't think it's worth starting work until `aspnet/entityframework` publishes `2.0.0-preview2`.  There have been a couple heavy BC breaking changes between `2.0.0-preview1` and `2.0.0-preview2`:

- Dissolve `Relational.Design` package https://github.com/aspnet/EntityFramework/issues/7004
- Target `netstandard2.0` instead of `nedstandard1.3` https://github.com/aspnet/EntityFramework/pull/8595

Really it's just a case of waiting until `aspnet/EntityFramework` settles down with the breaking changes.  Then we look at https://github.com/aspnet/EntityFramework/tree/dev/src/EFCore.SqlServer as a reference and have to update our side with all of the changes they made.

I can work on the `Relational` package, which will need to get done first.  Once that's done, we'll need somebody to roll the `Design` package into the `Relational` package so that scaffolding works again.  Any volunteers?

**Comment by caleblloyd** [comment]
I've begun work on this in a [2.0.0-wip branch on my fork](https://github.com/caleblloyd/Pomelo.EntityFrameworkCore.MySql/tree/2.0.0-wip), targeting the nightly `2.0.0-preview2-*` upstream entity framework libraries on MyGet

Once I have it compiling and tests passing locally, I'll merge it over.  Hopefully that will be in a couple weeks.

**Comment by mguinness** [comment]
@caleblloyd FWIW, if you encounter issues Diego Vega at MSFT suggested these avenues for assistance:

> If you happen to own an EF Core provider and run into any questions when updating your code for EF Core 2, please contact us. GitHub (https://github.com/aspnet/EntityFramework/issues), email or twitter @efmagicunicorns.

**Comment by Kagamine** [comment]
Need a 2.0.0-preview1 release

**Comment by caleblloyd** [comment]
@kagamine I'm skipping straight to EF.Relational 2.0.0-preview2. Too much BC breakage between 2.0.0-preview1 and 2.0.0-preview2

**Comment by Kagamine** [comment]
Ok, so how about create a dev branch and continous integrate it in MyGet?

**Comment by caleblloyd** [comment]
I'm trying to get it to compile in a branch on my fork right now. Lots of changes in 2.0.0 upstream, going to take me another week or two. I'll PR it over to a dev branch once I get it to compile.  In the meantime this is the branch on my fork:

https://github.com/caleblloyd/Pomelo.EntityFrameworkCore.MySql/tree/2.0.0-wip

**Comment by JonasKlose** [comment]
I've created a little bounty for this issue as I'd very much like to try out a stable version with .net core 2.0.

https://www.bountysource.com/issues/45444603-upgrade-to-2-0-0-preview

**Comment by caleblloyd** [comment]
@JonasKlose thanks. I'd like to encourage anyone who isn't already contributing to take a look at the [2.0.0 milestone](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/milestone/6)

The biggest area right now is scaffolding, which has undergone many updates and is in need of revamp: https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/289

We are making our 2.0 updates to the `dev` branch.  I've been chipping away at the `Relational` package and that's getting pretty close to being on par with where we are at 1.x today.

**Comment by Kagamine** [comment]
Will we release a 2.0-preview2 package?

**Comment by Kagamine** [comment]
https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql/2.0.0-preview2-10028-unstable

**Comment by Kagamine** [comment]
@JonasKlose Please cancel the bounty. We have already published an unstable release and the stable one will come soon. Thanks for your support.

**Comment by caleblloyd** [comment]
The first 2.0.0-preview2 has been released at [2.0.0-preview2-10043](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql/2.0.0-preview2-10043)

Please use build tag `2.0.0-preview2-*` or `2.0.0-*` to get new updates.

It should be a straightforward update from 1.x.  You need to NetStandard 2.0 support, you can get that at the [.NET Core 2.0 Preview](https://www.microsoft.com/net/core/preview) website for .NET Core or enable it in Visual Studio for .NET Framework 4.6.1 and up.  Note you must target `netcoreapp2.0` or `net461` or higher.

All of the EF tools also have to be brought up to `2.0.0-*`.  For an example refer to our [Functional Test .csproj](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/test/EFCore.MySql.FunctionalTests/EFCore.MySql.FunctionalTests.csproj)

Try it out and open any issues you may find!  We'd like to harden `2.0.0-preview` so that we're ready within a couple weeks after .NET Core 2.0 RTM hits.
